tests/test_param_routing.py

```diff
@@ -26,15 +26,18 @@ def test_routing_basic(wrapper, builder):
 
     foo_val = object()
 
-    est = wrapper(model=builder, model__hidden_layer_sizes=(100,), model__foo=foo_val)
+    # build once to get expected meta-parameters
+    expected_meta = (
+        wrapper(model=builder, model__hidden_layer_sizes=(100,)).fit(X, y)._meta
```

Comment:
:-1: why does this depend on a private attribute? I say hard code this (and below too).

``` python
expected_meta = {
    "n_features_in_",
    "X_dtype_",
    "y_dtype_",
    "X_shape_",
    "y_ndim_",
    "target_type_"
}
```

Reply:
Agreed. I'll swap over the the hardcoded set. It'll be nice too because in the future adding any meta attributes will require editing this test.